lib/templates.php

```diff
@@ -111,13 +111,18 @@ function gutenberg_register_wp_theme_taxonomy() {
 /**
  * Automatically set the theme meta for templates.
  *
- * @param array $post_id Template ID.
+ * @param array   $post_id Template ID.
+ * @param WP_POST $post Template post object.
+ * @param boolean $update Whether this is an existing template being updated.
  */
-function gutenberg_set_template_and_template_part_post_theme( $post_id ) {
+function gutenberg_set_template_and_template_part_post_theme( $post_id, $post, $update ) {
 	$themes = wp_get_post_terms( $post_id, 'wp_theme' );
 	if ( ! $themes ) {
 		wp_set_post_terms( $post_id, array( wp_get_theme()->get_stylesheet() ), 'wp_theme', true );
 	}
+	if ( $update && has_term( '_wp_is_original', 'wp_theme', $post_id ) ) {
+		wp_remove_object_terms( $post_id, '_wp_is_original', 'wp_theme' );
+	}
```

Comment:
I think this would wrongly remove the term if it's being updated based on a file change?